File: fragments/inheritance_refactoring/before.py
class Person(object):
    def __init__(self, age, job): 
        self.age = age
        self.job = job

    def birthday(self): 
        self.age += 1

class Pet(object):
    def __init__(self, age, owner): 
        self.age = age
        self.owner = owner

    def birthday(self): 
        self.age += 1

me=Person(37,"Programmer")
me.birthday()
assert(me.age == 38)

File: session06/python/before.cpp
#include <string>
using namespace std;

class Person{
    public:
    int age;
    string job;
    Person(int age, string job): 
        age(age), job(job)
        {}
};

class Pet{
    int age;
    Person & owner;
    Pet(int age, Person & owner):
        age(age), owner(owner)
        {}
};

int main(int argc, char **argv){
    Person me(37,"Programmer");
}
